.github/workflows/build.yaml

```diff
@@ -9,6 +9,76 @@ on:
     branches: [ master ]
 
 jobs:
+  changes:
+    name: changed files
+    runs-on: ubuntu-latest
+    # Set job outputs to values from changed_files step for other jobs
+    outputs:
+      codecheck: ${{ steps.changed_files.outputs.codecheck }}
+      clang_tidy: ${{ steps.changed_files.outputs.clang_tidy }}
+      run_test_suite: ${{ steps.changed_files.outputs.test_suite }}
+      inst_windows_msvc: ${{ steps.changed_files.outputs.inst_windows_msvc }}
+      inst_windows: ${{ steps.changed_files.outputs.inst_windows }}
+      inst_macos: ${{ steps.changed_files.outputs.inst_macos }}
+      appimage: ${{ steps.changed_files.outputs.appimage }}
+    steps:
+    - name: Check for changed files
+      uses: dorny/paths-filter@v2
+      id: changed_files
+      with:
+        # groups to check for changed files (as string, yaml anchors are resolved by the tool)
+        filters: |
+          codecheck:
+            - 'CMakeLists.txt'
+            - '*/CMakeLists.txt'
+            - 'cmake/codecheck/**'
+            - 'cmake/WlFunctions.cmake'
+            - 'src/**'
+            - '.github/scripts/install_deps.sh' # calls install-dependencies.sh
+            - '.github/workflows/build_codecheck.yaml'
+          clang_tidy:
+            - &src_cmake_dependencies
```

Comment:
Please create a dummy container filter, e.g. `common:` where `&src_cmake_dependencies` and `&installer` are defined, to make both the definitions and the first users more readable.

Reply:
There is some overhead to calculate the change for an unused filter, but it is neglectable.

Reply:
Actually the example of shared filters at `dorny/paths-filter` does just this: defines a dummy, even with the same name as the filter itself. If we put the 2 shared filters into the same dummy container, then we even spare one compared to that.

Reply:
This is pushed now.